release-calendar/src/client/api-service.ts

```diff
@@ -14,20 +14,37 @@
  * limitations under the License.
  */
 
-import {Release} from './models/view-models';
-import {Release as ReleaseEntity} from '../types';
+import {
+  ApiServiceInterface,
+  Promotion as PromotionEntity,
+  Release as ReleaseEntity,
+} from '../types';
+import {Promotion, Release} from './models/view-models';
 import fetch from 'node-fetch';
 const SERVER_URL = `http://localhost:3000`;
 
-export class ApiService implements ApiService {
-  private getRequest(url: string): Promise<ReleaseEntity[]> {
+export class ApiService implements ApiServiceInterface {
+  private getReleaseRequest(url: string): Promise<ReleaseEntity[]> {
+    return fetch(url).then((result) => result.json());
+  }
+
+  private getPromotionRequest(url: string): Promise<PromotionEntity[]> {
     return fetch(url).then((result) => result.json());
   }
 
   async getReleases(): Promise<Release[]> {
-    const releases = await this.getRequest(SERVER_URL);
+    const releases = await this.getReleaseRequest(SERVER_URL);
     return releases.map((release: ReleaseEntity) => {
       return new Release(release);
     });
   }
+
+  async getCurrentReleases(): Promise<Promotion[]> {
+    const currentReleases = await this.getPromotionRequest(
+      SERVER_URL + '/current-releases/',
+    );
+    return currentReleases.map((promotion: PromotionEntity) => {
+      return new Promotion(promotion);
```

Comment:
What if you converted `PromotionEntity` directly to `Map<Channel, string>`? That way you don't have to convert it twice.

Reply:
That's what I was doing originally but then i realized it wasn't following the view model system we had kind of set up. I can switch back. But just to make sure that would mean removing the `current-releases`/`promotion-current` file in `models`?

Reply:
Yeah. And you can keep the view model system by adding a class to the view model, not necessarily a `Promotion` but something like `CurrentReleases`.

You don't have to have a 1:1 match with view models and entities. View models can be whatever the client needs.